stripe/src/main/java/com/stripe/android/PaymentAuthConfig.kt

```diff
@@ -0,0 +1,612 @@
+package com.stripe.android
+
+import android.app.Activity
+import androidx.annotation.IntRange
+import com.stripe.android.stripe3ds2.init.ui.ButtonCustomization
+import com.stripe.android.stripe3ds2.init.ui.LabelCustomization
+import com.stripe.android.stripe3ds2.init.ui.StripeButtonCustomization
+import com.stripe.android.stripe3ds2.init.ui.StripeLabelCustomization
+import com.stripe.android.stripe3ds2.init.ui.StripeTextBoxCustomization
+import com.stripe.android.stripe3ds2.init.ui.StripeToolbarCustomization
+import com.stripe.android.stripe3ds2.init.ui.StripeUiCustomization
+import com.stripe.android.stripe3ds2.init.ui.TextBoxCustomization
+import com.stripe.android.stripe3ds2.init.ui.ToolbarCustomization
+import com.stripe.android.stripe3ds2.init.ui.UiCustomization
+
+/**
+ * Configuration for authentication mechanisms via [PaymentController]
+ */
+class PaymentAuthConfig private constructor(
+    internal val stripe3ds2Config: Stripe3ds2Config
+) {
+    class Builder : ObjectBuilder<PaymentAuthConfig> {
+        private var stripe3ds2Config: Stripe3ds2Config? = null
+
+        fun set3ds2Config(stripe3ds2Config: Stripe3ds2Config): Builder {
+            this.stripe3ds2Config = stripe3ds2Config
+            return this
+        }
+
+        override fun build(): PaymentAuthConfig {
+            return PaymentAuthConfig(requireNotNull(stripe3ds2Config))
+        }
+    }
+
+    data class Stripe3ds2Config internal constructor(
+        @IntRange(from = 5, to = 99) internal val timeout: Int,
+        internal val uiCustomization: Stripe3ds2UiCustomization
+    ) {
+        init {
+            checkValidTimeout(timeout)
+        }
+
+        private fun checkValidTimeout(timeout: Int) {
+            require(!(timeout < 5 || timeout > 99)) {
+                "Timeout value must be between 5 and 99, inclusive"
+            }
+        }
+
+        class Builder : ObjectBuilder<Stripe3ds2Config> {
+            private var timeout = DEFAULT_TIMEOUT
```

Comment:
would be helpful to mention the unit in a comment, i'm guessing this is in seconds?